### PR TITLE
Add TCP zero copy example to the TCP echo client demo - IPv6 WinSim

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR adds the TCP zero copy interface example to the TCP echo client demo IPv6 WinSim project.

Test Steps
-----------
Tested by connecting to an echo server both on HW (STM32) and simulator.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- ~[ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.~

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sirnish — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Review by ActoryOu — APPROVED** [review]

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/TCPEchoClient_SingleTasks.c [hunk 1]
```diff
@@ -70,6 +70,9 @@
 /* The number of instances of the echo client task to create. */
     #define echoNUM_ECHO_CLIENTS          ( 1 )
 
+/* To enable the use of zero copy interface of the TCP sockets */
+    #define USE_TCP_ZERO_COPY             ( 0 )
```
Comment (moninom1) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/TCPEchoClient_SingleTasks.c:74]: Should we add this flag to FreeRTISIPConfig.h?

Comment (tony-josi-aws) [reply FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/TCPEchoClient_SingleTasks.c:74]: Keeping this flag to just the demo file will help not to build the entire source files when just the zero copy macro is changed. Another reason is to make it similar to the convention we follow in existing demos:[ UDPEchoClient_SingleTasks.c: USE_ZERO_COPY](https://github.com/FreeRTOS/FreeRTOS/blob/0b5cd0866590b63346b95464a45186b74d35f916/FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/UDPEchoClient_SingleTasks.c#L53)

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/TCPEchoClient_SingleTasks.c [hunk 2]
```diff
@@ -151,6 +154,68 @@
...
+                 * that its a zero copy */
+                pcDstBuffer = NULL;
+            }
+        #endif /* USE_TCP_ZERO_COPY */
+
+        /* Send the string to the socket. */
+        lTransmitted = FreeRTOS_send( xSocket,                /* The socket being sent to. */
+                                      ( void * ) pcDstBuffer, /* The data being sent. */
+                                      lStringLength,          /* The length of the data being sent. */
+                                      0 );                    /* No flags. */
```
Comment (moninom1) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/TCPEchoClient_SingleTasks.c:212]: Do we need the ZERO_COPY flag here?

Comment (tony-josi-aws) [reply FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/TCPEchoClient_SingleTasks.c:212]: For TCP zero copy, `FreeRTOS_send` identifies its a zero copy by checking if the `pcDstBuffer` is NULL, hence there is no zero copy flag for `FreeRTOS_send`.

Comment (moninom1) [reply FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_IPv6_Demo/IPv6_Multi_WinSim_demo/TCPEchoClient_SingleTasks.c:212]: Got it Thanks

**Review by moninom1 — APPROVED** [review]

**Pull request merged**